File: Dal/Models/PresentationDay.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DAL.Models
{
    public class PresentationDay
    {
        public int PresentationDayId { get; set; }
        public DateTime Date { get; set; }

        public int ExamPeriodId { get; set; }
        public ExamPeriod ExamPeriod { get; set; } = null!;

        public ICollection<Slot> Slots { get; set; } = new List<Slot>();
    }
}

File: Client/src/app/shared/models/presentationday.ts
import { Slot } from './slot';

// Note: To avoid circular reference issues, use 'any' for ExamPeriod and Slot for now.
export interface PresentationDay {
  presentationDayId: number;
  date: string; // ISO string for Date
  examPeriodId: number;
  examPeriod: any;
  slots: Slot[];
}
